Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Services/JsonExportProvider.cs

```diff
@@ -12,20 +19,89 @@ public class JsonExportProvider : IExportProvider
         public IExportProviderConfiguration Configuration { get; }
         public ExportedTypeMetadata Metadata { get; set; }
 
+        private JsonSerializer Serializer { get; set; }
+        private StreamWriter Writer { get; set; }
+
+
         public JsonExportProvider(Stream stream, IExportProviderConfiguration exportProviderConfiguration)
         {
+
+            if (exportProviderConfiguration is JsonProviderConfiguration jsonProviderConfiguration)
+            {
+                Serializer = JsonSerializer.Create(jsonProviderConfiguration.Settings);
+            }
+            else
+            {
+                Serializer = JsonSerializer.CreateDefault();
+            }
+
             _stream = stream;
             Configuration = exportProviderConfiguration;
+
+            Writer = new StreamWriter(_stream);
         }
 
         public void WriteMetadata(ExportedTypeMetadata metadata)
         {
-            throw new System.NotImplementedException();
+
         }
 
         public void WriteRecord(object objectToRecord)
         {
-            throw new System.NotImplementedException();
+            FilterProperties(objectToRecord);
+            Serializer.Serialize(Writer, objectToRecord);
+            Writer.Flush();
+        }
+
+
+        private void FilterProperties(object obj, string baseMemberName = null)
+        {
+            var type = obj.GetType();
+
+            foreach (var property in type.GetProperties().Where(x => x.CanRead && x.CanWrite))
+            {
+                var propertyName = $"{baseMemberName}{(baseMemberName.IsNullOrEmpty() ? string.Empty : ".")}{property.Name}";
```

Comment:
Please check how we can reuse ExportedTypeMetadata code in this method logic - e.g string template should be defined only once. Probably ExportedTypeMetadata will need some refactoring for it.